From ff9cfea0e796b695ec0d53ea6a1b112a7126f73b Mon Sep 17 00:00:00 2001
From: Varun Sampat <vsampat@twitter.com>
Date: Thu, 23 Apr 2020 17:30:51 -0400
Subject: [PATCH] IPMB support for NM commands

---
 kcsbridged.cpp | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/kcsbridged.cpp b/kcsbridged.cpp
index a37cf5f..c4ce50f 100644
--- a/kcsbridged.cpp
+++ b/kcsbridged.cpp
@@ -81,6 +81,12 @@ class SmsChannel
     static constexpr size_t kcsMessageSize = 256;
     static constexpr uint8_t netFnShift = 2;
     static constexpr uint8_t lunMask = (1 << netFnShift) - 1;
+    static constexpr uint8_t netFnAppRes = 0x7;
+    static constexpr uint8_t cmdGetMsg = 0x33;
+    static constexpr uint8_t cmdSendMsg = 0x34;
+    static constexpr uint8_t ipmiCcOk = 0x0;
+
+
 
     SmsChannel(std::shared_ptr<boost::asio::io_context>& io,
                std::shared_ptr<sdbusplus::asio::connection>& bus,
@@ -224,6 +230,11 @@ class SmsChannel
                         entry("LUN=0x%02x", lun), entry("CMD=0x%02x", cmd),
                         entry("CC=0x%02x", cc));
                 }
+                if (netfn == netFnAppRes &&
+                    cmd == cmdGetMsg && cc == ipmiCcOk)
+                {
+                    clearAttention();
+                }
                 boost::system::error_code ecWr;
                 size_t wlen =
                     boost::asio::write(*dev, boost::asio::buffer(rsp), ecWr);
@@ -236,6 +247,11 @@ class SmsChannel
                         entry("NETFN=0x%02x", netfn), entry("LUN=0x%02x", lun),
                         entry("CMD=0x%02x", cmd), entry("CC=0x%02x", cc));
                 }
+                if (netfn ==  netFnAppRes && cmd == cmdSendMsg &&
+                    cc == ipmiCcOk && payload.size() == 0)
+                {
+                    setAttention();
+                }
             },
             ipmiQueueService, ipmiQueuePath, ipmiQueueIntf, ipmiQueueMethod,
             netfn, lun, cmd, data, options);
