From 50e5e9d12a1b944e36dd9c42bdf92fdb2a5c5f7a Mon Sep 17 00:00:00 2001
From: jeannie <jeannie@localhost.localdomain>
Date: Thu, 18 Apr 2019 18:15:28 +0800
Subject: [PATCH] Enhance command response buffer size

---
 ipmid.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ipmid.hpp b/ipmid.hpp
index 93a85b0..0e119dc 100644
--- a/ipmid.hpp
+++ b/ipmid.hpp
@@ -19,6 +19,6 @@ ipmi_ret_t ipmi_netfn_router(const ipmi_netfn_t, const ipmi_cmd_t,
 // The BT FIFO in the AST2400 can only handle 64 bytes.
 // Can only allow 63 because the BT interface still
 // needs 1 byte for the length field.
-#define MAX_IPMI_BUFFER 64
+#define MAX_IPMI_BUFFER 256
 
 extern FILE *ipmiio, *ipmidbus, *ipmicmddetails;
