From 9738356a4dd1d46826809a3187effdec80a3c249 Mon Sep 17 00:00:00 2001
From: Ray Lue <ray.lue@mic.com.tw>
Date: Tue, 4 Feb 2020 18:24:08 +0800
Subject: [PATCH] make pgood property writable

---
 libopenbmc_intf/openbmc_intf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/openbmc_intf.c b/openbmc_intf.c
index 32efdd8..30ff139 100644
--- a/openbmc_intf.c
+++ b/openbmc_intf.c
@@ -16273,7 +16273,7 @@ static const _ExtendedGDBusPropertyInfo _control_power_property_info_pgood =
     -1,
     (gchar *) "pgood",
     (gchar *) "i",
-    G_DBUS_PROPERTY_INFO_FLAGS_READABLE,
+    G_DBUS_PROPERTY_INFO_FLAGS_READABLE |G_DBUS_PROPERTY_INFO_FLAGS_WRITABLE,
     NULL
   },
   "pgood",
