From 6f87ce7855989c927d1220acb42603b4966cb126 Mon Sep 17 00:00:00 2001
From: John Chung <john.chung@mic.com.tw>
Date: Fri, 18 Sep 2020 14:38:44 +0800
Subject: [PATCH] Set the limit size for dbus

---
 bus/system.conf.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/bus/system.conf.in b/bus/system.conf.in
index f139b55..a5702f3 100644
--- a/bus/system.conf.in
+++ b/bus/system.conf.in
@@ -108,11 +108,11 @@
        133169152 bytes = 127 MiB
        33554432 bytes = 32 MiB
        150000ms = 2.5 minutes -->
-  <!-- <limit name="max_incoming_bytes">133169152</limit> -->
+  <limit name="max_incoming_bytes">33554432</limit>
   <!-- <limit name="max_incoming_unix_fds">64</limit> -->
-  <!-- <limit name="max_outgoing_bytes">133169152</limit> -->
+  <limit name="max_outgoing_bytes">33554432</limit>
   <!-- <limit name="max_outgoing_unix_fds">64</limit> -->
-  <!-- <limit name="max_message_size">33554432</limit> -->
+  <limit name="max_message_size">33554432</limit>
   <!-- <limit name="max_message_unix_fds">16</limit> -->
   <!-- <limit name="service_start_timeout">25000</limit> -->
   <!-- <limit name="auth_timeout">5000</limit> -->
