From 68520d024883e2fd9762aeee0929b61cae7655c0 Mon Sep 17 00:00:00 2001
From: OpenEmbedded <oe.patch@oe>
Date: Wed, 13 Jun 2018 17:29:27 +0800
Subject: [PATCH] Modified ibm-cffps driver

---
 drivers/hwmon/pmbus/ibm-cffps.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/hwmon/pmbus/ibm-cffps.c b/drivers/hwmon/pmbus/ibm-cffps.c
index 93d9a9e..86f9869 100644
--- a/drivers/hwmon/pmbus/ibm-cffps.c
+++ b/drivers/hwmon/pmbus/ibm-cffps.c
@@ -337,7 +337,10 @@ static struct pmbus_driver_info ibm_cffps_info = {
 	.func[0] = PMBUS_HAVE_VIN | PMBUS_HAVE_VOUT | PMBUS_HAVE_IOUT |
 		PMBUS_HAVE_PIN | PMBUS_HAVE_FAN12 | PMBUS_HAVE_TEMP |
 		PMBUS_HAVE_TEMP2 | PMBUS_HAVE_TEMP3 | PMBUS_HAVE_STATUS_VOUT |
-		PMBUS_HAVE_STATUS_IOUT | PMBUS_HAVE_STATUS_INPUT |
+/*Jeannie mod for support pout */
+//		PMBUS_HAVE_STATUS_IOUT | PMBUS_HAVE_STATUS_INPUT | 
+              PMBUS_HAVE_STATUS_IOUT | PMBUS_HAVE_STATUS_INPUT | PMBUS_HAVE_POUT |
+//Jeannie mod end
 		PMBUS_HAVE_STATUS_TEMP | PMBUS_HAVE_STATUS_FAN12,
 	.read_byte_data = ibm_cffps_read_byte_data,
 	.read_word_data = ibm_cffps_read_word_data,
-- 
2.7.4

