From 4f3c781ee8568a0997e9ab8f4b12f9e994553c05 Mon Sep 17 00:00:00 2001
From: Ray Lue <ray.lue@mic.com.tw>
Date: Mon, 11 May 2020 17:34:51 +0800
Subject: [PATCH 1/1] remove host state change in pgood monitor

---
 gpioMonMain.cpp | 35 +++++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)
 mode change 100644 => 100755 gpioMonMain.cpp

diff --git a/gpioMonMain.cpp b/gpioMonMain.cpp
old mode 100644
new mode 100755
index 251e0d5..d56417f
--- a/gpioMonMain.cpp
+++ b/gpioMonMain.cpp
@@ -41,6 +41,37 @@ std::map<std::string, int> polarityMap = {
 }
 } // namespace phosphor
 
+static constexpr char const* PgService = "org.openbmc.control.Power";
+static constexpr char const* PgObj = "/org/openbmc/control/power0";
+static constexpr char const* PgIntf = "org.openbmc.control.Power";
+#define PROPERTY_INTERFACE "org.freedesktop.DBus.Properties"
+
+void pgoodhook(gpiod_line* line)
+{
+
+    if (gpiod_line_request_input(line, "gpio_monitor") < 0)
+    {
+         std::cerr << "Error request pgood\n";
+         return;
+    }
+
+    int resp = gpiod_line_get_value(line);
+
+    gpiod_line_release(line); 
+    auto bus = sdbusplus::bus::new_default();
+    std::cerr << "pgood:" << resp <<"\n";
+    auto method = bus.new_method_call(PgService, PgObj, PROPERTY_INTERFACE, "Set");
+    method.append(PgIntf, "pgood", sdbusplus::message::variant<int>((int)resp));
+    bus.call_noreply(method);
+
+    method = bus.new_method_call(PgService, PgObj, PROPERTY_INTERFACE, "Set");
+    method.append(PgIntf, "state", sdbusplus::message::variant<int>((int)resp));
+    bus.call_noreply(method);
+    
+    std::cerr << "pgood state init to: " << (int)resp << "\n";
+
+}
+
 int main(int argc, char** argv)
 {
 
@@ -133,6 +164,10 @@ int main(int argc, char** argv)
             std::string lineName = obj["LineName"];
             lineMsg += lineName;
             line = gpiod_line_find(lineName.c_str());
+            if(lineName.compare("PS_PWROK") == 0)
+            {
+                pgoodhook(line);
+            }
         }
 
         if (line == NULL)
-- 
2.26.2

