From 68d4a2d55b0f0fb363cd2411abaafb3103d1ae1b Mon Sep 17 00:00:00 2001
From: Ray Lue <ray.lue@mic.com.tw>
Date: Tue, 10 Dec 2019 20:19:28 +0800
Subject: [PATCH 1/1] NVME change to use NVME-MI smbus address

---
 configurations/NVME P4000.json | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configurations/NVME P4000.json b/configurations/NVME P4000.json
index 46ea3f5..b2d6c83 100644
--- a/configurations/NVME P4000.json	
+++ b/configurations/NVME P4000.json	
@@ -8,7 +8,7 @@
             "Type": "EEPROM"
         },
         {
-            "Address": "0x53",
+            "Address": "0x6A",
             "Bus": "$bus",
             "Name": "NVMe $index Temp",
             "Thresholds": [
@@ -49,4 +49,4 @@
         "PartNumber": "$PRODUCT_PART_NUMBER",
         "SerialNumber": "$PRODUCT_SERIAL_NUMBER"
     }
-}
\ No newline at end of file
+}
-- 
2.22.0

