From 3f4d4479db1dc3815f2cc31e0d441ee1abada36d Mon Sep 17 00:00:00 2001
From: jeannie <jeannie@localhost.localdomain>
Date: Thu, 13 Jun 2019 15:06:59 +0800
Subject: [PATCH] Remove smbioshandler.cpp fro CMake config to avoid build
 error

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 630a696..aa5c3db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,7 +91,7 @@ include_directories (${OPENSSL_INCLUDE_DIR})
 add_library (
     zinteloemcmds SHARED src/oemcommands.cpp src/sensorcommands.cpp
     src/storagecommands.cpp src/multinodecommands.cpp src/firmware-update.cpp
-    src/smbioshandler.cpp src/smbiosmdrv2handler.cpp
+ #   src/smbioshandler.cpp src/smbiosmdrv2handler.cpp
     src/manufacturingcommands.cpp src/bmccontrolservices.cpp
     src/bridgingcommands.cpp
 )
