From 28365e35f883414e4647633b73c476f319caab3f Mon Sep 17 00:00:00 2001
From: Mirage Su <mirage.su@mic.com.tw>
Date: Tue, 26 Nov 2019 17:43:28 +0800
Subject: [PATCH] s7106 add OCP NIC temperature sensor

---
 src/HwmonTempMain.cpp | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/HwmonTempMain.cpp b/src/HwmonTempMain.cpp
index 98058b8..ad93312 100755
--- a/src/HwmonTempMain.cpp
+++ b/src/HwmonTempMain.cpp
@@ -195,10 +195,13 @@ void createSensors(
 
 		std::vector<thresholds::Threshold> t = sensorThresholds; 
 		
-        sensors[sensorName] = std::make_unique<HwmonTempSensor>(
-            directory.string() + "/temp1_input", sensorType, objectServer,
-            dbusConnection, io, sensorName, std::move(sensorThresholds),
-            *interfacePath);
+        if (deviceName != "14-001f")
+        {
+            sensors[sensorName] = std::make_unique<HwmonTempSensor>(
+                    directory.string() + "/temp1_input", sensorType, objectServer,
+                    dbusConnection, io, sensorName, std::move(sensorThresholds),
+                    *interfacePath);
+        }
         auto findSecondName = baseConfiguration->second.find("Name1");
         if (findSecondName == baseConfiguration->second.end())
         {
