From 176ff087aadee66a69af1d06f3e701b67d2e4416 Mon Sep 17 00:00:00 2001
From: Nik Johnson <njohnson@twitter.com>
Date: Wed, 15 Apr 2020 18:46:06 -0400
Subject: [PATCH] Write to SEL when reset button is pressed

---
 src/button_handler.cpp | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/src/button_handler.cpp b/src/button_handler.cpp
index 385bb4d..b28723e 100755
--- a/src/button_handler.cpp
+++ b/src/button_handler.cpp
@@ -218,6 +218,12 @@ void Handler::longPowerPressed(sdbusplus::message::message& msg)
 
 void Handler::resetPressed(sdbusplus::message::message& msg)
 {
+
+    uint16_t genid = 0x20;
+    bool assert = 1;
+    std::vector<uint8_t> eventData(selEvtDataMaxSize, 0xFF);
+    eventData[0] = 0x02;
+
     try
     {
         if (!poweredOn())
@@ -228,6 +234,18 @@ void Handler::resetPressed(sdbusplus::message::message& msg)
 
         log<level::INFO>("Handling reset button press");
 
+        sdbusplus::message::message writeSEL = bus.new_method_call(
+            ipmiSELService, ipmiSELPath, ipmiSELAddInterface, "IpmiSelAdd");
+        writeSEL.append(ipmiSELAddMessage, sensorPath, eventData, assert,
+                    genid);
+        try
+        {
+            bus.call(writeSEL);
+        }
+        catch (sdbusplus::exception_t& e)
+        {
+            log<level::ERR>("call IpmiSelAdd failed");
+        }
     }
     catch (SdBusError& e)
     {
