From 337f432279cd0717b982581788f68c39fce01dca Mon Sep 17 00:00:00 2001
From: "wendy.shi" <wendy.shi@mic.com.tw>
Date: Fri, 5 Jun 2020 11:58:17 +0800
Subject: [PATCH] Set the default UART route setting

---
 board/aspeed/ast-g5/ast-g5.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/board/aspeed/ast-g5/ast-g5.c b/board/aspeed/ast-g5/ast-g5.c
index e67a4bf..a7da3bf 100644
--- a/board/aspeed/ast-g5/ast-g5.c
+++ b/board/aspeed/ast-g5/ast-g5.c
@@ -16,11 +16,22 @@
 
 DECLARE_GLOBAL_DATA_PTR;
 
+/*--------------------------------------------------------------------
+ * @fn Set_Default_UART_Route
+ * @brief Set the default UART Route
+ *--------------------------------------------------------------------*/
+void Set_Default_UART_Route(void)
+{
+    puts("Set the default UART Route(UART1 <-> UART3 for SOL)...\n");
+    *(volatile ulong *)(0x1E78909C) = 0x1450000 ;
+}
+
 int board_init(void)
 {
 	gd->bd->bi_boot_params = CONFIG_SYS_SDRAM_BASE + 0x100;
 	gd->flags = 0;
 
+	Set_Default_UART_Route();
 	return 0;
 }
 
