From e86ba8887a4af1f77cb70a8beb15346bce1520c9 Mon Sep 17 00:00:00 2001
From: "wendy.shi" <wendy.shi@mic.com.tw>
Date: Thu, 4 Jun 2020 17:13:34 +0800
Subject: [PATCH] Set BMC RTL8211E PHY LED

---
 drivers/net/ftgmac100.c | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/drivers/net/ftgmac100.c b/drivers/net/ftgmac100.c
index 65b062a..2c62738 100644
--- a/drivers/net/ftgmac100.c
+++ b/drivers/net/ftgmac100.c
@@ -383,6 +383,30 @@ static int ftgmac100_phy_init(struct eth_device *dev)
 	return 1;
 }
 
+static int ftgmac100_phy_led_init(struct eth_device *dev)
+{
+	struct ftgmac100_data *priv = dev->priv;
+ 	//RTL8211E phy
+  	// Switch to LED control page
+  	ftgmac100_phy_write(dev, priv->phy_addr, 0x1f, 0x0007);
+  	ftgmac100_phy_write(dev, priv->phy_addr, 0x1e, 0x002c);
+
+  	// Write LED settings
+  	ftgmac100_phy_write(dev, priv->phy_addr, 0x1c, 0x0742);
+  	ftgmac100_phy_write(dev, priv->phy_addr, 0x1a, 0x0040);
+
+  	// Switch back to PHY default page
+  	ftgmac100_phy_write(dev, priv->phy_addr, 0x1f, 0x0000);
+
+  	/* Do not advertise EEE */
+ 	ftgmac100_phy_write(dev, priv->phy_addr, 0x0d, 0x0007);
+ 	ftgmac100_phy_write(dev, priv->phy_addr, 0x0e, 0x003c);
+  	ftgmac100_phy_write(dev, priv->phy_addr, 0x0d, 0x4007);
+	ftgmac100_phy_write(dev, priv->phy_addr, 0x0e, 0x0000);
+
+  return 0;
+}
+
 static int ftgmac100_update_link_speed(struct eth_device *dev)
 {
 	struct ftgmac100 *ftgmac100 = (struct ftgmac100 *)dev->iobase;
@@ -798,6 +822,10 @@ int ftgmac100_initialize(bd_t *bd)
 		ftgmac100_set_mac_from_env(dev);
 
 		card_number++;
+
+		ftgmac100_phy_init(dev);
+                ftgmac100_update_link_speed(dev);
+                ftgmac100_phy_led_init(dev);
 	}
 	return card_number;
 
