From 733806d406f822f47d385dd28cb88034f2686403 Mon Sep 17 00:00:00 2001
From: John Chung <john.chung@mic.com.tw>
Date: Mon, 6 Nov 2023 09:18:07 +0000
Subject: [PATCH 2/3] Disable uboot delay

---
 include/configs/ast-common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/ast-common.h b/include/configs/ast-common.h
index a0243083bd..25d3d97cef 100644
--- a/include/configs/ast-common.h
+++ b/include/configs/ast-common.h
@@ -70,7 +70,7 @@
 /*
  * Environment Config
  */
-#define CONFIG_BOOTDELAY		2
+#define CONFIG_BOOTDELAY		0
 
 /*
  * Miscellaneous configurable options
-- 
2.25.1

