From 1fea2e2f931f7f90aad61e30683702a14dda88af Mon Sep 17 00:00:00 2001
From: Ray Lue <ray.lue@mic.com.tw>
Date: Mon, 6 Jan 2020 17:51:08 +0800
Subject: [PATCH 1/1] fix ipmi get sel issue

---
 src/sensorcommands.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/sensorcommands.cpp b/src/sensorcommands.cpp
index f06bce7..939c83f 100755
--- a/src/sensorcommands.cpp
+++ b/src/sensorcommands.cpp
@@ -1196,7 +1196,7 @@ static void loadSdr(uint16_t baseID)
     tmp.header.record_length = sizeof(SensorDataEventOnlyRecord) -
                                       sizeof(get_sdr::SensorDataRecordHeader); 
     tmp.key.owner_lun = 0x0;
-    tmp.body.entity_id = 0x0;
+    tmp.body.entity_id = 0x07;
     tmp.body.entity_instance = 0x01;
     tmp.body.event_reading_type = 0x6f;
     
@@ -1365,7 +1365,7 @@ ipmi::RspType<uint16_t,            // next record ID
         record.key.owner_lun = 0x0;
         record.key.sensor_number = sensornumber;
 
-        record.body.entity_id = 0x0;
+        record.body.entity_id = 0x07;
         record.body.entity_instance = 0x01;
         record.body.sensor_type = getSensorTypeFromPath(path);
         record.body.event_reading_type = 0x6f;
@@ -1421,7 +1421,7 @@ ipmi::RspType<uint16_t,            // next record ID
     record.key.owner_lun = 0x0;
     record.key.sensor_number = sensornumber;
 
-    record.body.entity_id = 0x0;
+    record.body.entity_id = 0x7;
     record.body.entity_instance = 0x01;
     record.body.sensor_capabilities = 0x68; // auto rearm - todo hysteresis
     record.body.sensor_type = getSensorTypeFromPath(path);
-- 
2.22.0

