From f6409ae737e92a6aed0655d3665b7220209dd40c Mon Sep 17 00:00:00 2001
From: John Chung <john.chung@mic.com.tw>
Date: Wed, 23 Aug 2023 12:22:23 +0000
Subject: [PATCH] Set VGA source with AC lost status setting

Signed-off-by: John Chung <john.chung@mic.com.tw>
---
 board/aspeed/ast-g5/ast-g5.c | 26 +++++++++++++++++++++++++-
 1 file changed, 25 insertions(+), 1 deletion(-)

diff --git a/board/aspeed/ast-g5/ast-g5.c b/board/aspeed/ast-g5/ast-g5.c
index eebd4a7edb..ccd8db1f55 100755
--- a/board/aspeed/ast-g5/ast-g5.c
+++ b/board/aspeed/ast-g5/ast-g5.c
@@ -98,6 +98,29 @@ void Set_HW_Watchdog(void)
     writel(0x13, AST_WDT2_BASE + 0x0c);
 }
 
+/*--------------------------------------------------------------------
+ * @fn Set_VGA_source
+ * @brief Set VGA source with AC lost status setting
+ *--------------------------------------------------------------------*/
+void Set_VGA_source(void)
+{
+    puts("Set the default  VGA source setting: ");
+
+    // Check BMC AC lost
+    if (ast_scu_read(AST_SCU_SYS_CTRL) & (0x1)){
+        puts ("Graphics CRT mode\n");
+        *(volatile ulong *)(AST_SCU_BASE + AST_SCU_MISC1_CTRL) |= 0x00010000;
+        *(volatile ulong *)(0x1e723100) = 0x41434C09; //ascii "A C L (checksum)"
+    }
+    else
+    {
+        puts ("VGA mode\n");
+        *(volatile ulong *)(AST_SCU_BASE + AST_SCU_MISC1_CTRL) &= 0xFFFCFFFF;
+    }
+
+}
+
+
 int board_init(void)
 {
 	gd->bd->bi_boot_params = CONFIG_SYS_SDRAM_BASE + 0x100;
@@ -107,7 +130,8 @@ int board_init(void)
     Set_Default_UART_Route();
 	Set_Default_GPIO_Pass_Through();
 	Set_HW_Watchdog();
-    
+    Set_VGA_source();
+
 	return 0;
 }
 
-- 
2.25.1

