From 8bf954d9941cfe0de99a671dd80ea72d7842a707 Mon Sep 17 00:00:00 2001
From: "ray.lue" <ray.lue@mic.com.tw>
Date: Wed, 17 Apr 2019 17:49:19 +0800
Subject: [PATCH] force bus#3 to be 8-bit EEPROM

---
 src/FruDevice.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/FruDevice.cpp b/src/FruDevice.cpp
index a3936c6..5ff3e20 100755
--- a/src/FruDevice.cpp
+++ b/src/FruDevice.cpp
@@ -39,7 +39,7 @@ extern "C" {
 }
 
 namespace fs = std::filesystem;
-static constexpr bool DEBUG = false;
+static constexpr bool DEBUG = true;
 static size_t UNKNOWN_BUS_OBJECT_COUNT = 0;
 constexpr size_t MAX_FRU_SIZE = 512;
 constexpr size_t MAX_EEPROM_PAGE_INDEX = 255;
@@ -154,6 +154,7 @@ int get_bus_frus(int file, int first, int last, int bus,
             /* Check for Device type if it is 8 bit or 16 bit */
             //int flag = isDevice16Bit(file);
 			int flag = 1;
+			if(bus == 3) flag = 0;
 			
             if (flag < 0)
             {
