From 51de40560825347697e7b561529783537799517e Mon Sep 17 00:00:00 2001
From: Nik Johnson <njohnson@twitter.com>
Date: Mon, 27 Jan 2020 16:57:27 -0500
Subject: [PATCH] FRU state sensor

---
 include/sdrutils.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/sdrutils.hpp b/include/sdrutils.hpp
index 30302df..e98227d 100755
--- a/include/sdrutils.hpp
+++ b/include/sdrutils.hpp
@@ -169,6 +169,7 @@ enum class SensorTypeCodes : uint8_t
     acpi = 0x22,
     mgtsubsyshealth = 0x28,
     versionchange = 0x2b,
+    fru_state = 0x2c,
     other = 0xB,
 };

@@ -186,6 +187,7 @@ const static boost::container::flat_map<const char*, SensorTypeCodes, CmpStr>
                  {"acpi", SensorTypeCodes::acpi},     
                  {"mgtsubsyshealth", SensorTypeCodes::mgtsubsyshealth}, 
                  {"versionchange", SensorTypeCodes::versionchange},     
+                 {"fru_state", SensorTypeCodes::fru_state},
                  {"power", SensorTypeCodes::other}}};
 
 inline static std::string getSensorTypeStringFromPath(const std::string& path)
