From 2c6865d0a9fe6add72952e80a7e8942a4ca09c48 Mon Sep 17 00:00:00 2001
From: John Chung <john.chung@mic.com.tw>
Date: Tue, 21 Jul 2020 19:39:35 +0800
Subject: [PATCH] Add debug message for watchdog reset

---
 app/watchdog.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/app/watchdog.cpp b/app/watchdog.cpp
index fca27b7..d87c82d 100755
--- a/app/watchdog.cpp
+++ b/app/watchdog.cpp
@@ -58,6 +58,7 @@ ipmi::RspType<> ipmiAppResetWatchdogTimer()
         // The ipmi standard dictates we enable the watchdog during reset
         wd_service.resetTimeRemaining(true);
         lastCallSuccessful = true;
+        std::cerr << "Reset watchdog timer and start counting.\n";
         return ipmi::responseSuccess();
     }
     catch (const InternalFailure& e)
-- 
2.7.4

